### PR TITLE
Add migrate support to `pallet-cosmwasm` and contracts

### PR DESCRIPTION
**Pull request by aeryz**
## Description
* Add `migrate` extrinsic for contract migration.
* Add `upload_and_instantiate` extrinsic for uploading and instantiating in a single contract.
* Add `migrate` entrypoint to all of the cosmwasm contracts as well as CONTRACT_VERSION and NAMES to make sure that migration is securely done.

## Checklist

- [ ] I have updated the cargo docs to reflect changes made by this PR _(if applicable)_
- [ ] I have updated the `book/` to reflect changes made by this PR _(if applicable)_

### CODE REVIEW

File: .github/workflows/build-test-deploy.yml [hunk 1]
```diff
@@ -554,8 +554,6 @@ jobs:
       - run: |
           cachix watch-exec -j 16 -c 0 composable-community nix -- build .#junod --no-update-lock-file --show-trace -L
           cachix watch-exec -j 16 -c 0 composable-community nix -- build .#gex --no-update-lock-file --show-trace -L
-          cachix watch-exec -j 16 -c 0 composable-community nix -- build .#wasmswap --no-update-lock-file --show-trace -L
-
```
[the remark this answers is not in the document]
Comment (aeryz) [reply .github/workflows/build-test-deploy.yml:558]: I think this comes from the other PR. cc @dzmitry-lahoda @hussein-aitlahcen

File: .nix/composable-bin.nix [hunk 2]
```diff
@@ -1,27 +0,0 @@
...
-    mv release/composable $out/bin
-    mv doc $out
-  '';
-  ROCKSDB_LIB_DIR = "${pkgs.rocksdb}/lib";
-  LD_LIBRARY_PATH = pkgs.lib.strings.makeLibraryPath [
-    pkgs.stdenv.cc.cc.lib
-    pkgs.llvmPackages.libclang.lib
-    pkgs.rocksdb
-  ];
-}
```
[the remark this answers is not in the document]
Comment (aeryz) [reply .nix/composable-bin.nix:27]: this as well

File: frame/cosmwasm/src/lib.rs [hunk 3]
```diff
@@ -443,6 +445,40 @@ pub mod pallet {
...
+			code: ContractCodeOf<T>,
+			salt: ContractSaltOf<T>,
+			admin: Option<AccountIdOf<T>>,
+			label: ContractLabelOf<T>,
+			funds: FundsOf<T>,
+			gas: u64,
+			message: ContractMessageOf<T>,
+		) -> DispatchResultWithPostInfo {
+			let next_code_id =
+				CurrentCodeId::<T>::get().checked_add(1).ok_or(Error::<T>::CodeIdOverflow)?;
```
Comment (KaiserKarel) [inline frame/cosmwasm/src/lib.rs:478]: You can use the Nonce abstraction for this?

**Comment by dzmitry-lahoda** [comment]
i reverted changes about running devnets from  branch i put them. afaik in comsoverse we are on that branch so wanted cleaup, not on main. it creates complication tohave cached builds reuse while we are on that branch in cosmoverse.

i did not modified this pe and not aware what it is doing.

File: frame/cosmwasm/src/lib.rs (hunk 3 above)
Comment (aeryz) [reply frame/cosmwasm/src/lib.rs:478]: We removed this extrinsic, instead we introduce a new identifier for code. Which is an `enum code_id | code_hash`. This way users won't need to wait for the code id to be generated and returned.

**Pull request closed without merging**